shared/test/metabase/shared/models/visualization_settings_test.cljc

```diff
@@ -0,0 +1,51 @@
+(ns metabase.shared.models.visualization-settings-test
+  "Tests for the shared visualization-settings namespace functions"
+  #?@
+      (:clj
+       [(:require [clojure.test :as t]
+                  [metabase.shared.models.visualization-settings :as mb.viz])]
+       :cljs
+       [(:require [clojure.test :as t]
+                  [goog.string :as gstring]
+                  [metabase.shared.models.visualization-settings :as mb.viz])]))
+
+(def fmt #?(:clj format :cljs gstring/format))
+
+(t/deftest parse-column-ref-strings-test
+  (t/testing "Column ref strings are parsed correctly"
+    (let [f-qual-nm "/databases/MY_DB/tables/MY_TBL/fields/COL1"
+          f-id      42
+          col-nm    "Year"]
+      (doseq [[input-str expected] [[(fmt "[\"ref\",[\"field\",%d,null]]" f-id) {::mb.viz/field-id f-id}]
```

Comment:
Two things:
- Does this handle other field ids in the options? e.g. `["field", 123, {"source-field": 456}]`
- Pivot tables (and maybe other places) have lists of field refs that aren't prefixed with "ref".

Reply:
I think I will need to gather these examples to beef up this test (even if they aren't immediately relevant for serialization).